presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -11,13 +11,15 @@ Overview
 --------
 
 This connector allows the use of `Apache Kafka <https://kafka.apache.org/>`_
-topics as tables in Presto.
-Each message is presented as a row in Presto.
+topics as tables in Presto. Each message is presented as a row in Presto.
 
 Topics can be live. Rows appear as data arrives, and disappear as
 segments get dropped. This can result in strange behavior if accessing the
 same table multiple times in a single query (e.g., performing a self join).
 
+The connector reads data from partitioned topics in parallel across workers to
```

Comment:
Topic is always partioned (sometimes it has just a single partition). Presto also reads single partition in parallel. Single split contains by default 10000 messages (it is configurable).

Reply:
```
    @Config("kafka.messages-per-split")
    @ConfigDescription("Count of Kafka messages to be processed by single Presto Kafka connector split")
```
It defaults to 100_000.
So if Kafka topic has 2 partitions where one has 50_000 and other had 150_000 messages then they will be processed by 3 tasks in Presto.